Add support for output directory in extraction (#16126)
coq/coq: doc/sphinx/addendum/extraction.rst

   If this :term:`flag` is set, fully expand Coq types in ML.  See the Coq source code to learn more.

.. opt:: Extraction Output Directory @string

   Sets the directory in which the extracted `.ml` (and `.mli`) files will be
   written. The default is the current directory which can be found using

>    :cmd:`Pwd`.

Reviewer: Does it take a relative or absolute path?

Author: It can take both. Of course, relative is probably more sensible.

>    Sets the directory in which the extracted `.ml` (and `.mli`) files will be
>    written. The default is the current directory which can be found using
>    :cmd:`Pwd`.

Reviewer 2: ```suggestion
   Sets the directory where extracted `.ml` and `.mli` files will be
   written. The default is the current directory, which can be displayed with
   :cmd:`Pwd`.
```
